Guava TableAssert: assertions for columnCount, rowCount, size

mk868 opened the pull request:
Hello,

In this PR, I add two methods that allow for asserting the count of columns and rows:

- `rowCount()`
- `columnCount()`
- `size()`

Related suggestion: https://github.com/assertj/assertj/pull/3606#issuecomment-2351045476

#### Check List:
* Unit tests : YES
* Javadoc with a code example (on API only) : YES
* PR meets the [contributing guidelines](https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md)

joel-costigliola commented:
@mk868 in order to be able to return to the table assertion, could you follow what we did for `Map` size assertion, see MapSizeAssert and AbstractMapSizeAssert ?

Thanks !

mk868 commented:
@joel-costigliola
Thank you for tips!
My changes:

- I added assertions for the size of the table (number of declared cells in the table): `size()`
- I created a `TableIntegerAssert` that is reused for columnCount/rowCount/size assertions (If it is too general, let me know, I will split it into `AbstractTableIntegerAssert`. `TableRowCountAssert`, `TableColumnCountAssert`. `TableSizeAssert`)
- added `@CheckReturnValue` annotations

mk868 commented:
Hi,
Do you have any suggestions for this PR?
Are these changes appropriate to merge?
